tests/cpp_caching/YRoot

```diff
@@ -0,0 +1,66 @@
+ExtDockerImage('ubuntu', image='ubuntu', tag='18.04')
+
+AptPackage('gnupg', package='gnupg')
+
+AptPackage('curl', package='curl')
+
+AptPackage('ca-certificates', package='ca-certificates')
+
+CustomInstaller(
+    'llvm-repo-key',
+    uri='',
+    deps=[':curl', ':ca-certificates',],
+    script='install-llvm-repo-key.sh',
+)
+
+DockerImage('ubuntu-gpg', base_image=':ubuntu', deps=':gnupg')
+
+AptGroup(
+    'clang',
+    packages=[
+        'clang-8',
+        'libclang-common-8-dev',
+        'libclang-8-dev',
+        'libllvm8',
+        'lldb-8',
+        'llvm-8',
+        'llvm-8-dev',
+        'llvm-8-runtime',
+    ],
+    deps=[':llvm-repo-key'],
+#    repository='deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-8 main',
+#    repo_key='15CF4D18AF4F7421'
+)
+
+CustomInstaller(
+    'gtest',
+    fetch='https://github.com/google/googletest/archive/release-1.8.0.tar.gz',
+    script='install-gtest.sh'
+)
+
+DockerImage('builder', base_image=':ubuntu-gpg', deps=':clang')
+
+DockerImage('builder-with-gtest', base_image=':builder', deps=':gtest')
```

Comment:
can all this be simplified to the bare minimum needed for the test?

definite no need for gtest

maybe would be more minimal to use g++ over clang in terms of all the boilerplate and dependencies

Reply:
but it's the same solution to prog and test, and they are different types of targets

Reply:
I added a test for GTest.
I looked at it with @dlehavi-resonai and we decided not to change it to g++ at the moment